proposals/0079-allow_commission_decrease_at_any_time.md

```diff
@@ -0,0 +1,83 @@
+---
+simd: '0079'
+title: Allow Commission Decrease at Any Time
+authors:
+  - Bryan Ischo (bryan@ischo.com)
+category: Standard
+type: Core
+status: Draft
+created: 2023-10-26
+feature: TBD
+---
+
+## Summary
+
+The commission_updates_only_allowed_in_first_half_of_epoch feature disallows
+commission decrease in the second half of epoch. Given that the purpose of this
+feature was to prevent 'rug pulls' which are accomplished by increasing
+commission at the end of epoch and then decreasing commission at the beginning
+of next epoch, disallowing decreases during the second half of epoch is
+unnecessary.
+
+A feature gate must be added to support this SIMD as all validators' vote
+programs must treat commission change instructions the same or else consensus
+will diverge.
+
+## Motivation
+
+Some validator operators may need to decrease commission in order to satisfy
+their own operational criteria.  Speaking personally, when I have an operator
+error that results in reduced stake account rewards for the epoch, I reduce
+commission to 0% to ensure that stake accounts are not disadvantaged by my
+actions. Not being allowed to do this in the second half of an epoch is a real
+problem for me.
+
+Allowing commission changes during the second half of epochs also enables
+"reverse rug pulls" where a validator has commission set to 100% during the
+majority of an epoch but reduces it to 0% across the epoch boundary.  It's not
+clear why some validator operators choose to do this, but it does not seem
+harmful, and there does appear to be motivation for some validators to do this
+as it's been done numerous times in the past.
+
+## Alternatives Considered
+
+No alternatives were considered.
+
+## New Terminology
+
+Reverse Rug Pull: Setting commission to 100% for most of an epoch, and to 0%
+only across epoch boundaries.  This has the effect of discouraging third party
+stake to a validator while still charging 0% commission.  This is not considered
+to be a technique harmful to stakers.
+
+## Detailed Design
+
+A feature will be added which, when enabled, should cause all node
+implementations' vote program's set-commission instruction handling to first
+check whether the proposed commission change is a decrease, and if so, do not
+invoke the "only allow commission change in first half of epoch" rule.
```

Comment:
How about relaxing it a tiny bit more and also allowing validators to set the same commission that is currently set? Ie. allowing any number `<=` current commission instead of just `<`

Reply:
I will do this, but I don't understand the motivation.  Allowing the commission to be set to the same value is a no-op, and I don't know why allowing no-ops to succeed has value.  I think it would be more valuable for no-ops to fail as an indication to whoever submitted the tx that they probably didn't do what they *thought* they were trying to do.

Reply:
In some cases that definitely makes sense, but I think your point addresses a separate issue than this SIMD, which is about restoring previous behavior in certain cases.

If we wanted to implement the fail-on-noop behavior, then my take would be to *always* do it. But that would probably be better addressed in a separate SIMD, which I am *not* advocating for right now